assets/js/components/HostDetails/HostSettingsPage.jsx

```diff
@@ -21,16 +23,36 @@ function HostSettingsPage() {
     loading: catalogLoading,
   } = useSelector(getCatalog());
 
+  const { saving } = useSelector(getCheckSelection());
+
   if (!host) {
     return <LoadingBox text="Loading..." />;
   }
+
   const {
     hostname: hostName,
     provider,
     agent_version: agentVersion,
     selected_checks: selectedChecks,
   } = host;
 
+  const refreshCatalog = () =>
+    dispatch(
+      updateCatalog({
+        provider: host.provider,
+        target_type: 'host',
+      })
+    );
+
+  const saveSelection = (selection, targetID) =>
```

Comment:
I already mentioned, but I think `hostName` should come as arg.
Actually you have it in `HostCheckSelection`, so it would be free.

Reply:
d89913dfd016ccfd45ce79493562594576c85bd4